docs/rules/no-for-loop-extra-variable.md

```diff
@@ -0,0 +1,43 @@
+# Do not declare an extra variable in a `for` loop init
+
+A pattern in some codebases is to declare a second variable in the initial state which is then only used in the test condition. This variable is unecessary and prevents the `no-for-loop` rule from flagging the loop.
+
+In the case where the second variable is used, either inside the loop, or is defined with `var` and used elsewhere in the code, it often makes more sense to define the variable outside the loop so that the intent of the loop is easier to understand.
```

Comment:
It wouldn't make sense to define the variable with `let` and use it in other places in the code because the scope of the variable would be contained to the for loop. If using `var` then the variable can be used in other places.